src/main/java/cafe/cryptography/curve25519/CompressedEdwardsY.java

```diff
@@ -32,39 +32,22 @@ public CompressedEdwardsY(byte[] data) {
      * @return an EdwardsPoint, if this is a valid encoding.
      */
     public EdwardsPoint decompress() {
-        FieldElement X, Y, YY, u, v, v3, vXX, check;
-        Y = FieldElement.fromByteArray(data);
-        YY = Y.square();
+        FieldElement Y = FieldElement.fromByteArray(data);
+        FieldElement YY = Y.square();
 
         // u = y²-1
-        u = YY.subtract(FieldElement.ONE);
+        FieldElement u = YY.subtract(FieldElement.ONE);
 
         // v = dy²+1
-        v = YY.multiply(Constants.EDWARDS_D).add(FieldElement.ONE);
+        FieldElement v = YY.multiply(Constants.EDWARDS_D).add(FieldElement.ONE);
 
-        // v3 = v³
-        v3 = v.square().multiply(v);
-
-        // x = (v3²)vu, aka x = uv⁷
-        X = v3.square().multiply(v).multiply(u);
-
-        // x = (uv⁷)^((q-5)/8)
-        X = X.pow22523();
-
-        // x = uv³(uv⁷)^((q-5)/8)
-        X = v3.multiply(u).multiply(X);
-
-        vXX = X.square().multiply(v);
-        check = vXX.subtract(u); // vx²-u
-        if (check.isZero() == 0) {
-            check = vXX.add(u); // vx²+u
-            if (check.isZero() == 0) {
-                throw new IllegalArgumentException("not a valid EdwardsPoint");
-            }
-            X = X.multiply(Constants.SQRT_M1);
+        FieldElement.SqrtRatioM1 sqrt = FieldElement.sqrtRatioM1(u, v);
```

Comment:
FieldElement.SqrtRationM1 and FieldElement.sqrtRatioM1 are very close. FieldElement.SqrtRatioM1Result?